src/main/java/hudson/plugins/release/pipeline/ReleaseStep.java

```diff
@@ -0,0 +1,218 @@
+import hudson.model.TaskListener;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.plugins.release.ReleaseWrapper;
+import hudson.plugins.release.SafeParametersAction;
+import jenkins.model.Jenkins;
+import jenkins.model.ParameterizedJobMixIn;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Created by e3cmea on 1/3/17.
+ *
+ * @author Alexey Merezhin
+ */
+public class ReleaseStep extends AbstractStepImpl {
+    private static final Logger LOGGER = Logger.getLogger(ReleaseStep.class.getName());
+
+    private String job;
+    private List<ParameterValue> parameters;
+
+    @DataBoundConstructor
+    public ReleaseStep(String job) {
+        this.job = job;
+        parameters = new ArrayList<>();
+    }
+
+    public String getJob() {
+        return job;
+    }
+
+    public void setJob(String job) {
+        this.job = job;
+    }
+
+    public List<ParameterValue> getParameters() {
+        return parameters;
+    }
+
+    @DataBoundSetter public void setParameters(List<ParameterValue> parameters) {
+        this.parameters = parameters;
+    }
+
+    public static class Execution extends AbstractStepExecutionImpl {
+        @StepContextParameter private transient Run<?,?> invokingRun;
+        @StepContextParameter private transient TaskListener listener;
+
+        @Inject(optional=true) transient ReleaseStep step;
+
+        private List<ParameterValue> updateParametersWithDefaults(AbstractProject project,
+                List<ParameterValue> parameters) throws AbortException {
+
+            if (project instanceof BuildableItemWithBuildWrappers) {
+                ReleaseWrapper wrapper = ((BuildableItemWithBuildWrappers) project).getBuildWrappersList()
+                                                                                   .get(ReleaseWrapper.class);
+                if (wrapper != null) {
+                    for (ParameterDefinition pd : wrapper.getParameterDefinitions()) {
+                        boolean parameterExists = false;
+                        for (ParameterValue pv : parameters) {
+                            if (pv.getName()
+                                  .equals(pd.getName())) {
+                                parameterExists = true;
+                                break;
+                            }
+                        }
+                        if (!parameterExists) {
+                            parameters.add(pd.getDefaultParameterValue());
+                        }
+                    }
+                } else {
+                    throw new AbortException("Job doesn't have release plugin configuration");
+                }
+            }
+            return parameters;
+        }
+
+        @Override
+        public boolean start() throws Exception {
+            if (step.getJob() == null) {
+                throw new AbortException("Job name is not defined.");
+            }
+
+            final AbstractProject project = Jenkins.getActiveInstance()
```

Comment:
🐛 for using `AbstractProject` in the new code. It won't support Pipeline in such way. There is a `ParameterizedJobMixIn.ParameterizedJob`if you need to process all jobs with parameters